Merge back 'chore_release-8.5.0' into 'chore_release-pd-8.5.0'

ddcc4 opened the pull request:
Here we go again :)

I'll wait for @mjhuff's PR #18709, which @jerader wants to propagate to `edge` for the Labware Library release.

ecormany committed:
docs(api): resolve TODO for 2.23 changes to InstrumentContext.name (#18702)

# Overview

See here's the thing about TODOs: you actually have _to do_ them.

thx @ddcc4 

## Test Plan and Hands on Testing

there's [a
sandbox](http://sandbox.docs.opentrons.com/pipette-name-todo-finally/v2/new_protocol_api.html#opentrons.protocol_api.InstrumentContext.name)

## Changelog

uncommented the thing we were supposed to a while ago!

## Risk assessment

negative

sfoster1 committed:
fix(hardware-testing): fix hanging gravi analysis (#18685)

You can't really use click.test's isolated environments because it needs
to be able to import hardware testing. It also needs to clean up its
recording thread on failure in addition to on success, and finally doing
these thigns reveals some failures we'll also have to fix.

---------

Co-authored-by: Ryan Howard <ryan.howard@opentrons.com>

mjhuff committed:
refactor(app): round pipetting command params to 2 sig figs (#18708)

Closes RQA-3937

Rounds all pipetting command params to two significant figures.

mjhuff committed:
refactor(shared-data): update lid def compatibility (#18709)

mjhuff reviewed (approved):
App & shared-data LGTM. Thank you!

sanni-t committed:
fix(api): require LC transfers to use API v2.24 (#18712)

Closes AUTH-2004

# Overview

Updates the liquid classes based transfer, consolidate and distribute
methods to require 2.24 instead of 2.23. Details in ticket above for why
we need to do this.

## Risk assessment

Not really. We had previously indicated in api source code that liquid
classes work is in-progress and subject to change. And these methods are
not available in public docs until 8.5.0.
